dask/dataframe/io.py

```diff
@@ -25,6 +25,33 @@
 
 csv_defaults = {'compression': None}
 
+
+def try_pd_read_csv(*args, **kwargs):
+    msg = """
+    By inspecting the first 1,000 rows of your csv file we guessed that the
+    column: %s had dtype: %s. But now that we're reading all the data we see it
+    has some values with dtype: %s. You should probably add the following
+    keyword arguments to your `read_csv` call:
+
+        dtype={'%s': %s}
+    """
+
+    try:
+        return pd.read_csv(*args, **kwargs)
+    except ValueError as e:
+        try:
+            # this is brittle
+            e_parts = e.message.split(' ')
+            column_number = int(e_parts[-1])
+            column_name = kwargs.get('names')[column_number]
+            dtype1 = e_parts[7]
+            dtype2 = e_parts[9]
+            msg %= (column_name, dtype1, dtype2, column_name, dtype2)
+        except:
+            raise e
```

Comment:
It is raising the exception from the level above this.

So if parsing the pandas exception raises an exception. The pandas exception will get raised, instead of an exception from parsing the exception.